docs/contributing.rst

```diff
@@ -47,6 +52,67 @@ We can also install ``tox`` to run the full test suite which also covers things
 
 It is important to understand that each Pull Request must pass the full test suite as part of the CI check, hence it is often convenient to have ``tox`` installed locally as well.
 
+Type Hints
+~~~~~~~~~~
+
+The code bases is transitioning to use `type hints <https://www.python.org/dev/peps/pep-0484/>`_. Type hints make it easy to prevent certain types of bugs, enable richer tooling and enhance the documentation, making the code easier to follow.
+
+All new code is required to land with type hints with the exception of test code that is not expected to use type hints.
+
+All parameters as well as the return type of defs are expected to be typed with the exception of ``self`` and ``cls`` as seen in the following example.
+
+.. code:: python
+
+    def __init__(self, wrapped_db: BaseDB) -> None:
+        self.wrapped_db = wrapped_db
+        self.reset()
+
+Documentation
+~~~~~~~~~~~~~
+
+Public APIs are expected to be annotated with docstrings as seen in the following example.
+
+.. code:: python
```

Comment:
Cool, if this highlights everything correctly!

I thought it had to be `.. code-block:: python` based on my reading of sphinx docs: http://www.sphinx-doc.org/en/stable/markup/code.html